### DISCUSSION TITLE
Captions

### DISCUSSION THREAD

**Discussion by therealharshhacker** [libDrive/libDrive#15] [Ideas]
Hi , first of all thanks for this amazing project!
Also I would like to suggest an ability to add subtitles for the movies or shows ..

1. User can select the subtitles themselves.
2. We can put .srt subtitles in the drive so that they automatically load up

**Comment by eliasbenb** [comment]
This is pretty much the same as issue #9
But you want the ability to add your own subtitles.

The first part is doable.
Second part is also doable, but I won't do it because it will make metadata gathering extremely slow

**Reply by therealharshhacker** [reply]
I don't really know much about how metadata is being gathered , but I think you shouldn't gather the subtitles when rebuilding metadata.
Instead when the video is played the code just checks if a .srt file exists with the same name as the video. If yes it loads it.

Anyways , I am looking forward for the first part .
Thanks

**Comment by eliasbenb** [comment]
I just pushed out a commit which implements part 1
https://github.com/libDrive/web/commit/7dbb0a9e0f4239db5b3972617e9c151475f7e2e0

You can view a preview here: <https://1b2741ca.libdrive-web.pages.dev> (you need to login with your own server)

Here's a screenshot: 

![image](https://user-images.githubusercontent.com/54410649/119169379-fd154600-ba72-11eb-97eb-a6a8d7f82f28.png)

The third button prompts the user to choose a file.

**Reply by therealharshhacker** [reply]
Thanks I am really grateful!

**Comment by eliasbenb** [comment]
Update:

Added support TV shows too

https://github.com/libDrive/web/commit/8da3f3c2d562aae30191eaf559d5960b22a08cdb

New preview site: <https://d1e2c765.libdrive-web.pages.dev>

**Comment by eliasbenb** [comment]
I forgot to mention, SRT files will not work. HTML5 only supports VTT

**Reply by therealharshhacker** [reply]
Thanks !
I can use a convertor 👍

**Comment by eliasbenb** [comment]
Ignore my last comment, while HTML doesn't natively support SRT subtitles, I can manually convert the SRT to a VTT file. I just implemented this.

https://github.com/libDrive/web/commit/bca49fada4dd6d90102e8aeb1b864113bf42da43

Live preview here: <https://9c287f20.libdrive-web.pages.dev>
